contentcuration/contentcuration/frontend/channelEdit/components/edit/CategoryOptions.vue

```diff
@@ -0,0 +1,262 @@
+
+  /*
+   * This is used to create the categories object from which the dropdown list is generated
+   * and is the same as this to make sure the order in Kolibri and Studio are the same:
+   * https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/learn/assets/
+   *   src/views/CategorySearchModal/CategorySearchOptions.vue#L73
+   */
+  for (let subjectKey of Object.entries(Categories)
+    .sort((a, b) => a[1].length - b[1].length)
+    .map(a => a[0])) {
+    const ids = Categories[subjectKey].split('.');
+
+    let path = '';
+    let nested = categoriesObj;
+    for (let fragment of ids) {
+      path += fragment;
+      if (availablePaths[path]) {
+        const nestedKey = CategoriesLookup[path];
+        if (!nested[nestedKey]) {
+          nested[nestedKey] = {
+            value: path,
+            nested: {},
+          };
+        }
+        nested = nested[nestedKey].nested;
+        path += '.';
+      } else {
+        break;
+      }
+    }
+  }
+
+  const flattenCategories = obj => {
+    Object.keys(obj).forEach(key => {
+      if (key !== 'value' && key !== 'nested') {
+        dropdown.push({
+          text: key,
+          value: obj[key]['value'],
+        });
+      }
+      if (typeof obj[key] === 'object') {
+        flattenCategories(obj[key]);
+      }
+    });
+    return dropdown;
+  };
+
+  flattenCategories(categoriesObj);
+
+  export default {
+    name: 'CategoryOptions',
+    mixins: [constantsTranslationMixin, metadataTranslationMixin],
+    props: {
+      value: {
+        type: Array,
+        default: () => [],
+      },
+    },
+    data() {
+      return {
+        categoryText: null,
+        chipsSelected: [],
+        nested: true,
+      };
+    },
+    computed: {
+      categoriesList() {
+        return dropdown.map(category => {
+          return {
+            ...category,
+            text: this.translateMetadataString(camelCase(category.text)),
+            level: this.findDepth(Categories[category.text]),
+          };
+        });
+      },
+      dropdownSelected() {
+        return [
+          ...new Set(
+            this.chipsSelected.reduce((a, chip) => a.concat(this.findFamilyTreeIds(chip)), [])
+          ),
+        ];
+      },
+      selected: {
+        get() {
+          return this.value;
+        },
+        set(value) {
+          let items = [...value];
+          let currentItem = items[items.length - 1] || '';
+
+          this.handleCheckboxItem(currentItem, items);
+          this.$emit('input', this.chipsSelected);
+        },
+      },
+      dropdownSelectedObj() {
+        return this.dropdownSelected.map(item => {
+          return {
+            text: this.translateMetadataString(camelCase(CategoriesLookup[item])),
+            value: item,
+            level: this.findDepth(item),
+          };
+        });
+      },
+    },
+    watch: {
+      categoryText(val) {
+        return !val ? (this.nested = true) : (this.nested = false);
```

Comment:
Could you add a short comment to this function to explain what it is doing? I think I understand, but it took me a couple of times reading through to follow